config/v1/types_oauth.go

```diff
@@ -1,37 +1,518 @@
 package v1
 
-import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OAuth Server and Identity Provider Config
 
 // +genclient
 // +genclient:nonNamespaced
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // OAuth holds cluster-wide information about OAuth.  The canonical name is `cluster`
-// TODO this object is an example of a possible grouping and is subject to change or removal
 type OAuth struct {
-	metav1.TypeMeta `json:",inline"`
-	// Standard object's metadata.
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	// spec holds user settable values for configuration
-	Spec OAuthSpec `json:"spec"`
-	// status holds observed values from the cluster. They may not be overridden.
-	Status OAuthStatus `json:"status"`
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              OAuthSpec   `json:"spec"`
+	Status            OAuthStatus `json:"status,omitempty"`
 }
 
+// OAuthSpec contains desired cluster auth configuration
 type OAuthSpec struct {
-	// options for configuring the embedded oauth server.
-	// possibly wellknown?
+	//IdentityProviders is an ordered list of ways for a user to identify themselves
+	IdentityProviders []IdentityProvider `json:"identityProviders"`
+
+	// TokenConfig contains options for authorization and access tokens
+	TokenConfig TokenConfig `json:"tokenConfig"`
+
+	// Templates allow you to customize pages like the login page.
+	// +optional
+	Templates OAuthTemplates `json:"templates"`
 }
 
+// OAuthStatus shows current known state of OAuth server in the cluster
 type OAuthStatus struct {
+	// TODO Fill in
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// TokenConfig holds the necessary configuration options for authorization and access tokens
+type TokenConfig struct {
+	// AuthorizeTokenMaxAgeSeconds defines the maximum age of authorize tokens
```

Comment:
sorry I didn't catch it before, but doc inside of the types.go is based on the serialization, not the member name so that when we generate the doc it matches for a user. that makes this `// authorizeTokenMaxAgeSeconds defines the maximum...`

Reply:
wat. ugh ok. will change everything